lib/util/stream_utils.js

```diff
@@ -87,21 +149,33 @@ shaka.util.StreamUtils = class {
         lowestAverageBandwidth = averageBandwidth;
       }
     });
-    goog.asserts.assert(bestCodecs != null, 'Should have chosen codecs!');
+
+    goog.asserts.assert(bestCodecs !== '', 'Should have chosen codecs!');
     goog.asserts.assert(!isNaN(lowestAverageBandwidth),
         'Bandwidth should be a number!');
 
-    // Filter out any variants that don't match, forcing AbrManager to choose
-    // from the most efficient variants possible.
-    manifest.variants = manifest.variants.filter((variant) => {
-      const codecs = variantCodecs(variant);
-      if (codecs == bestCodecs) {
-        return true;
-      }
+    return bestCodecs;
+  }
 
-      shaka.log.debug('Dropping Variant (better codec available)', variant);
-      return false;
-    });
+  /**
+   * Get group of variant codecs.
```

Comment:
I think a better description might be something like "Get a string representing all codecs used in a variant".